### PR TITLE
Fix for upgrading mapping

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*

*Description of changes:*

This PR fixes two bugs of upgrading mapping:
First, previously we only tried once when upgrading mapping.  It is possible we need to try a few more times.  The drawback is that if the code has a bug, we retry endlessly.  I should fix it in the future.
Second, we enclose the upgrade mapping function call under a user context in a recent commit.  Upgrade mapping will fail when security plugin is enabled since a regular user cannot upgrade mappings of system indices.

The PR also adds upper bounds to our dynamic settings if reasonable.
Testing done:
1. Tested that upgrade mapping succeeds with new changes.
2. Tested the upper bounds of the changed settings are in effect.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saratvemulapalli — APPROVED** [review]
Thanks for the change, didnt realize the update mapping would fail.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java [hunk 1]
```diff
@@ -578,7 +578,11 @@ public void updateMappingIfNecessary() {
         }
 
         final GroupedActionListener<Void> conglomerateListeneer = new GroupedActionListener<>(
-            ActionListener.wrap(r -> updateRunning.set(false), exception -> logger.error("Fail to updatea all mappings")),
+            ActionListener.wrap(r -> updateRunning.set(false), exception -> {
+                // TODO: don't retry endlessly. Can be annoying if there are too many exception logs.
+                updateRunning.set(false);
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java:583]: If remove this line, will not update index mapping if any exception happens?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java [hunk 2]
```diff
@@ -578,7 +578,11 @@ public void updateMappingIfNecessary() {
...
+                logger.error("Fail to updatea all mappings");
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java:584]: Add ad to the error message.

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java:584]: added

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/indices/AnomalyDetectionIndices.java:583]: if removing this line, we will only try to update mapping once per node.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**